### ISSUE TITLE
如何在loss.py文件中访问到当前的epoch值，参数如何传递呢？

### ISSUE THREAD

**Issue by dyhdesu** [ultralytics/ultralytics#19564]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

我想在yolov8的项目中，修改loss函数，已经增加了NWD函数，在计算loss函数的部分，希望能访问到当前的epoch值，使得loss计算跟当前epoch值有一个动态关联的关系。

通过default.yaml文件我知道了如何访问到其中的epochs参数值，但是不知道从哪获取当前的epoch值

### Additional

<img width="614" alt="Image" src="https://github.com/user-attachments/assets/fa80ee95-a7ff-4d40-bd99-644d0556bd76" />
<img width="742" alt="Image" src="https://github.com/user-attachments/assets/072dffce-b4b1-48a7-88be-db77db383a35" />
<img width="656" alt="Image" src="https://github.com/user-attachments/assets/690b1b1c-68b1-406a-bd97-35730aa1c11e" />

**Comment by UltralyticsAssistant** [comment]
👋 你好 @dyhdesu，感谢您对 Ultralytics 的兴趣 🚀！强烈建议您可以浏览我们的 [文档中心](https://docs.ultralytics.com/)，这里提供了丰富的 [Python](https://docs.ultralytics.com/usage/python/) 和 [CLI](https://docs.ultralytics.com/usage/cli/) 示例，同时也能解答许多常见的问题。

如果这是一个 🐛 Bug 报告，请提供一个[最小可复现的示例（MRE）](https://docs.ultralytics.com/help/minimum-reproducible-example/)，便于我们协助排查问题。

如果这是一个自定义训练相关的❓问题，请尽可能提供更多信息，例如您的数据集样例图片、训练日志，并确认已遵循我们的[最佳训练实践指南](https://docs.ultralytics.com/guides/model-training-tips/)。

加入 Ultralytics 社区，找到最适合您的交流方式！实时聊天可以加入我们的 [Discord 社区](https://discord.com/invite/ultralytics) 🎧。如果您喜欢深入讨论，可以访问 [Discourse 论坛](https://community.ultralytics.com/)。更喜欢分享和探索？来关注我们的 [Subreddit 社区](https://reddit.com/r/Ultralytics) 吧！

## 更新

请确保您已更新到最新的 `ultralytics` 软件包，并在 [**Python>=3.8**](https://www.python.org/) 和 [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) 环境中安装所有[依赖项](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml)，以验证您的问题是否已通过最新版本解决：

```bash
pip install -U ultralytics
```

## 环境

Ultralytics 的 YOLO 可以运行在以下最新验证的环境中（预装所有依赖项，包括 [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn)、[Python](https://www.python.org/) 和 [PyTorch](https://pytorch.org/)）：

- **Notebooks** with free GPU（免费 GPU 环境）：<a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. 查看 [GCP 快速入门指南](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. 查看 [AWS 快速入门指南](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker 镜像**. 查看 [Docker 快速入门指南](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## 状态

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

当上方的徽章显示绿色时，表示 Ultralytics 的 [CI 测试](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) 正常运行。CI 测试每日和每次提交对 macOS、Windows 和 Ubuntu 系统的所有 YOLO [模式](https://docs.ultralytics.com/modes/)和[任务](https://docs.ultralytics.com/tasks/)的正确性进行验证。

⚠️ 请注意，这是一个自动回复，Ultralytics 的工程师会尽快协助您解决问题 🙂

**Comment by glenn-jocher** [comment]
@dyhdesu 要在`loss.py`中获取当前epoch值，可以通过模型实例访问Trainer中的`epoch`属性。具体方法如下：

1. **在自定义Trainer中传递当前epoch**：
```python
# 在自定义Trainer类中添加
class CustomTrainer(DetectionTrainer):
    def get_model(self, cfg, weights):
        model = super().get_model(cfg, weights)
        model.trainer = self  # 将trainer实例绑定到model
        return model
```

2. **在loss计算中访问epoch**：
```python
# 在loss.py的损失函数中
current_epoch = self.model.trainer.epoch  # 直接访问当前epoch
```

这种方法通过模型实例的trainer属性链式访问到当前训练epoch。详细实现原理可参考[Ultralytics Engine Trainer文档](https://docs.ultralytics.com/reference/engine/trainer/)中关于训练循环和属性传递的部分。
